pandas/tests/arithmetic/test_string.py

```diff
@@ -261,10 +258,15 @@ def test_add_strings(any_string_dtype, request):
 
 
 @pytest.mark.xfail(reason="GH-28527")
-def test_add_frame(dtype):
+def test_add_frame(request, dtype):
+    if dtype.storage == "python":
+        # Inconsistent behavior between different versions of the python engine.
+        # Some return correctly, some return but with the wrong dtype
+        # Others just fail, we are blanket failing all
+        mark = pytest.mark.xfail(reason="[XPASS(strict)] GH-28527")
```

Comment:
this is confusing.  There is a decorator xfailing the test.  Now you're adding a second xfail with a reason being that it passes?  do two xfail markers cancel out?

i think the thing to do is remove the decorator and add the marker in only the actually-failing cases